src/argilla/client/feedback/schemas/questions.py

```diff
@@ -153,9 +154,26 @@ def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             ]
         if isinstance(values.get("labels"), list):
             values["settings"]["options"] = [{"value": label, "text": label} for label in values.get("labels")]
-        values["settings"]["visible_options"] = values.get(
-            "visible_labels"
-        )  # `None` is a possible value, which means all labels are visible
+        visible_options = values.get("visible_labels")
+        total_options = len(values["settings"]["options"])
+        if visible_options and visible_options > total_options:
+            if total_options >= 3:
+                warnings.warn(
+                    f"`visible_labels={visible_options}` is greater than the total number"
+                    f" of labels ({total_options}), so it will be set to `{total_options}`.",
```

Comment:
This message can apply to whenever the `visible_label` is set to a value greater than `total_options`. Not sure to understand the `3`-number's logic

Reply:
So I did it like this in case we allow modifications in the labels themselves, as we do for the moment with the `visible_options` field, so that the warning is directly related to the actual number of labels. Also because we set the minimum to 3 recently, and setting 3 may not be correct if we want to expand labels, at least that was what I thought, but maybe it's not necessary.

Reply:
Yes. This logic is okay.